src/oop/user_defined_classes/hard/q7/MainTest.java

```diff
@@ -0,0 +1,117 @@
+                Arguments.of("Master's Thesis", 50),
+                Arguments.of("Basic Life Skillz", 0),
+                Arguments.of("Still Life Painting", 10)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void courseConstructorInitializesAttributesCorrectly(String attribute1, int attribute2) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, String.class},
+                {attribute2, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectAttributeInstantiationMessage =
+                "Your " + testClassName + " constructor does not correctly initialize the object.";
+        _assertEquals(attribute1, classInstance.getFieldValue(instance, stringAttributeName1),
+                incorrectAttributeInstantiationMessage);
+        _assertEquals(attribute2, classInstance.getFieldValue(instance, intAttributeName1),
+                incorrectAttributeInstantiationMessage);
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of("Artificial Intelligence 3", 4, 3, "Artificial Intelligence 3 is a difficult course"),
+                Arguments.of("Matrix Algebra", 2, 10, "Matrix Algebra is not a difficult course"),
+                Arguments.of("Renaissance History", 1, 1, "Renaissance History is a difficult course"),
+                Arguments.of("Recreating Jurassic Park for Fun and Profit", 800, 3, "Recreating Jurassic Park for Fun and Profit is a difficult course"),
+                Arguments.of("Clown Studies", 5, 1000, "Clown Studies is not a difficult course"),
+                Arguments.of("Nepotism", -1, 0, "Nepotism is not a difficult course"),
+                Arguments.of("Learn to Blink", -50, -25, "Learn to Blink is not a difficult course")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    public void correctDetermineDifficultyMethod(String attribute1, int attribute2, int dif, String mssg) throws Throwable {
+        Object[][] arguments = {
+                {attribute1, String.class},
+                {attribute2, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        assertTrue(classInstance.hasMethod(methodName, new Class[]{int.class}, Void.TYPE, new String[]{"public"}),
+                "Your " + testClassName + " class is missing the method " + methodName + " or it does not have the correct return type or visibility modifier.");
+        classInstance.callMethod(methodName, new Object[][]{{dif, int.class}}, null, instance, new Clause[]{
+                new StringLiteral(mssg),
+                new Optional(new StringLiteral(" "))
+        }, "Your " + methodName + " method does not produce the correct output.");
```

Comment:
Also, could you change this to be more specific to the question? Maybe something about not printing the correct difficulty message.